pinot-common/src/main/java/org/apache/pinot/common/assignment/InstanceAssignmentConfigUtils.java

```diff
@@ -105,9 +106,11 @@ public static InstanceAssignmentConfig getInstanceAssignmentConfig(TableConfig t
     int numReplicaGroups = segmentConfig.getReplicationNumber();
     ReplicaGroupStrategyConfig replicaGroupStrategyConfig = segmentConfig.getReplicaGroupStrategyConfig();
     Preconditions.checkState(replicaGroupStrategyConfig != null, "Failed to find the replica-group strategy config");
+    SegmentPartitionConfig segmentPartitionConfig = tableConfig.getIndexingConfig().getSegmentPartitionConfig();
```

Comment:
So for tables which have legacy replica group assignment strategy configured, instance assignment will fail if they don't set partition config ? 

Our tooling that creates the actual table config hasn't yet migrated to new InstanceAssignmentReplicaGroupConfig. So, with this change it looks like we will fail do instance assignment for any non-partitioned replica group based table (new at the time of table creation) or existing table if rebalancer is invoked with reassignInstances=true

`Preconditions.checkState(segmentPartitionConfig != null, "Failed to find the segment partition config");` -> this is a breaking change. Why are we doing this ?

Reply:
I think I missed the case of  using `ReplicaGroupStrategyConfig` for non-partitioned table (the config key `numInstancesPerPartition` is confusing). Will revise the PR accordingly so that it is backward compatible

Reply:
Thanks. I don't think anything needs to be changed here. There are 3 scenarios:

- If the table is not partitioned - from legacy config, `numInstancesPerPartition` becomes `numInstancesPerReplicaGroup` in the new way of doing things. 

- If the table is partitioned but not partitioned at the replica group level - partitionColumn in legacy config is null.  This is same as previous scenario because replica group config does not get impacted

- If the table is partitioned and also partitioned at the replica group level - partitionColumn in legacy config is non-null. `numInstancesPerPartition` from legacy config is retained in the new config. `numPartitions` comes from `segmentPartitionConfig`. numInstancesPerReplicaGroup does not matter.

These are already handled.

Reply:
Correct. In order to use partitioned replica-group assignment, `ReplicaGroupStrategyConfig` is still required. We cannot directly take `SegmentPartitionConfig` because in certain scenarios, user might not want to use partitioned replica-group assignment even if segments are partitioned. Closing the PR